src/cmap/connection_pool.ts

```diff
@@ -432,18 +439,72 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
       return;
     }
 
+    const oldGeneration = this[kGeneration];
```

Comment:
Just to confirm - the ask is simply to move line 442 below the comment?

Reply:
yes, and to move the lines at the top of the function into the loadbalanced comment so that we could eventually separate these functions entirely; actually separating them into two separate functions does not have to happen here (but could, if you were so inclined, the pros are that it would make this separation of concerns cleaner, the cons is that it's a somewhat bigger refactor)